### DISCUSSION TITLE
WIF: Connect failed as no IP address received

### DISCUSSION THREAD

**Discussion by robespo72** [arendst/Tasmota#21608] [Q&A]
Hi Guys!

I have an ESP01 device equipped with Tasmota 12.5.0. (Core/SDK Version 2_7_4_9/2.2.2-dev) positioned quite close to the router and with excellent signal (RSSI:90,Signal:-55). It is periodically not reachable via web console, in fact having recorded its MAC Address (3C:61:05:CF:FD:D8) on an excel sheet I also see from the router that it is not connected, and this is compliant.

At this point I decided to connect the device via serial/UART and what I see from the Termite console is the following trace that is continuously shown on the video in an infinite loop (see picture below):

"WIF: Connect failed as no IP address received"

However in a completely arbitrary way every now and then it comes out of stuck and comes back online, only to then go back stuck again in times that are never well defined. If I remove the power and put it back causing a reboot it works again, but after a random time it goes back to stuck.

Is the state in which it ends (no IP address received) plausible? And if so, how come it systematically gets the IP address and works again as soon as it is rebooted?
This state instead could be a misleading indication while the real problem could be another? For example high temperatures? And what could be the correct wifipower setting for an ESP01?
Has it already happened to anyone of you? And how did you solve it?

I thank you in advance for all the information you can give me.
![ESP01_stuck](https://github.com/arendst/Tasmota/assets/151676414/8d2e5710-c882-45d3-995d-5b50fb52b656)

**Comment by arendst** [comment]
You could
- chk your router's DHCP server for free address range.
- updating to latest Tasmota version

**Comment by Jason2866** [comment]
Such strange issues with an esp8266 are often related to not good enough power supply.
ESP01 are known to be extremely picky here. If you do a search in Internet you will find many hits.

**Comment by robespo72** [comment]
Goodmorning to everyone!

I took a few days to create some media contents.
Honestly I don't think that the router's DHCP works not well, and I say this because of these two reasons:

1) if I restart the ESP01 it immediately obtains the IP, and this happens systematically at every reboot (see video "serial_log_ESP01_15-06-2024_11.00.mp4" and log "serial_log_ESP01_15-06-2024.log"). From the logs you can see how immediately after the reboot the ESP01 gets the IP 192.168.1.98;
2) the DHCP has been distributing IP addresses to all the devices in the house for a few years now, all the devices that connect to this home network obtain an IP without issues

As for the power supply, it seems quite stable around 3.3 vdc (see picture "ESP01_3.3_voltage_supply.jpg")

Another question I ask to the community is if there is a setting of tasmota that allows it to autonomously exit the infinite loop of "waiting for an IP address". In fact this state does not seem likely, in other words I don't think it's actually waiting for an IP address in that state since the IP address is easily obtained when requested (see after reboot).

Sure, I will be able to upgrade the Tasmota version to a later version than 13.0.
Furthermore, I also was wondering if there is an optimal wifipower value for the ESP01.

https://github.com/arendst/Tasmota/assets/151676414/530528c4-3c9e-4c94-bdc4-a707e62cdf4a

![ESP01_3 3_voltage_supply](https://github.com/arendst/Tasmota/assets/151676414/c86343ae-f054-4eac-8c11-3d53a3ff2866)
[serial_log_ESP01_15-06-2024.log](https://github.com/user-attachments/files/15846403/serial_log_ESP01_15-06-2024.log)

**Comment by Jason2866** [comment]
You will not see the short voltage drop downs happening with a multimeter. For this you need an oscilloscope.
I am quiet sure it IS a power supply issue. There is nowhere a decent cap in the 3.3V rail on the module, to soften/filter the peaks. To say it clear the ESP01 modules are a design fail at all.

**Reply by robespo72** [reply]
Firstly thanks for the suggestion Jason2866! I have already asked some of my friends, one of them has an oscilloscope, it will be interesting to find out because I use a 12vdc power supply whose manufacturer claims to be stabilized... but to be honest, I wouldn't be surprised if it wasn't. I will organize with this friend of mine and we will do this test.

So is there a relationship between the voltage spikes the ESP01 suffers from and the Tasmota status "WIF: Connection failed because no IP address was received"?

Apart from this test, I wanted to return to my question. Because if there is an answer then it would be appreciated because I would learn something more about the Tasmota world, if there is still no answer then the question could also be a small suggestion to further improve Tasmota which I consider a truly excellent software. And the question is this:
If Tasmota is confident of being in the "no IP address received" state, there is currently no configuration that can be set initially to tell it "if you are in this waiting state, every so many seconds exit this state and renew the request to obtain an IP from the router" etc. etc.
If, on the other hand, this setting does not currently exist then it could be hypothesized as an improvement to overcome the stuck.
On the other hand, the test shows that upon reboot it immediately obtains the IP, and in my experience this is systematic, so there is a small probability, which must be verified, that it is enough to request the IP again (during the reboot I suppose this request is made, although together with hundreds of other things) to unblock the situation and get out of the stuck.

**Reply by Jason2866** [reply]
The issue you encounter is directly related to power issue. The wifi "hardware" needs a lot of current. If this current is not delivered the results coming from the wifi part is garbage. Like you see with "no ip address"
To be clear nothing can be done in Tasmota software to heal unpredictable results getting from not working hardware part. Even worse you don't know what is maybe wrong. There will be no changes in Tasmota code for trying to fix a hardware issue.

The 12V power supply is one part. The mandantory part is the onboard 3.3V power regulator.
You have to check the 3.3V directly on the esp8266 pin(s).

**Reply by Jason2866** [reply]
Beside of this the Relais module you show in your picture is the worst one to get. It is cheapest in every aspect. If you want a working solution out of the box, buy a different one where the esp is mounted on the board. Every! relais board is better than this one. I know what I am talking about. Bought that cheap thing in my early Tasmota days too. Without my 40 years of experience in electronics i wouldn't have managed to get it going. The power issue problem is only one of them.

**Reply by robespo72** [reply]
Ok, I trust in your experience, it is not worth reinventing the wheel. I'm just gonna assume that the ESP01 and also the relay module are of poor quality, so I have already changed direction. I want to take advantage of your experience Jason2866, I changed everything, instead of the ESP01 I took a WemosD1Mini from AZDlv bla bla, a step-down module to transform the 12VDC voltage to 5VDC and a standalone relay.
The upstream power supply is provided by a stabilized power supply with 12VDC output up to 10A.
With this setting do you think I will have fewer problems?
In your opinion, what is the ideal setting?
Could the Wemos' antenna have current shortage problems even in this setting?
For any voltage spikes, what could be the solution instead?
![StepDown_WemosAZD_RelayMod](https://github.com/arendst/Tasmota/assets/151676414/597ce352-659f-4d4a-9833-dcdab4f37d1a)

**Reply by Jason2866** [reply]
The setup should work without issues. For reliable daily use, do not use Dupont wires. Solder everything.

**Reply by robespo72** [reply]
Hi Jason2866!

this sentence of yours piqued my curiosity:
The wifi "hardware" needs a lot of current. If this current is not delivered the results coming from the wifi part is garbage. Like you see with "no ip address"

Is there a power consumption graph that shows how much current an esp8266 uses while the hardware wifi is being managed?

**Reply by Jason2866** [reply]
https://www.ondrovo.com/a/20170207-esp-consumption/

**Reply by robespo72** [reply]
What are some good "voltage supervisors" for a WemosD1Mini that you have experienced with?
I mean those (maybe more than one??) with a good quality/price ratio

**Reply by Jason2866** [reply]
Replace the onboard LDO chip with a 500mA type. For example the ME6211C33 or a similar one which is rated for 500mA.

**Reply by robespo72** [reply]
Perfect! I'll do some tests on the bench, I guess I understood that I have to replace the LDO regulator on the relay shield with the ME6211C33 (or something similar to it).
But is it worth doing the same test on the Wemos D1 Mini? I mean .. replace the RT9013 LDO regulator of the WDM with the ME6211C33?

**Reply by Jason2866** [reply]
An orig. RT9013 LDO has 500mA. Replacement would be only needed if it is a fake / counterfeit one.
Most WemosD1 mini clones have counterfeit ones or use types not rated for 500mA.
I replaced all on my WemosD1 mini clones. No troubles since doing this.

**Comment by lalo-uy** [comment]
Or power the Wemos with a good 3.3V power supply.

**Reply by robespo72** [reply]
Thank you Ialo-uy!
It seems like an excellent solution to me, so let me see if I got this straight. if I use the 3.3VDC voltage supply the onboard LDO is bypassed, so I don't have to worry about whether it is good quality or not ... although I'm testing it now on some Wemos AZDblabla. To be honest I had never tried it because on the relay-shield that I use it clearly says 5VDC. So I've always used a 12VDC stabilized power supply, a step-down module to transform the voltage from 12 to 5 volts and with these I powered both the Wemos and the relay-shield. But now I downloaded the datasheet of the relay-shield, and it seems that it can be powered from 3 to 5 volts. So using the step-down to transform from 12 to 3.3 volts and with these powering both the Wemos and the relay-shield is fantastic. In fact, when put to on test it works fine. Does everything make sense to you?
Is this a better solution than the 5vdc voltage supply?
Do you think there are contraindications for this solution?
![voltage_supply_3 3vdc](https://github.com/arendst/Tasmota/assets/151676414/9b075679-6a30-4d7c-815d-5289e95e6382)

**Comment by lalo-uy** [comment]
In the pic I can’t read the whole relay JQC-3FF…..
The datasheet for the 5v version states the pickup voltag to 3.8v, so at 3.3 it is working out of spec.

**Reply by robespo72** [reply]
you're right, sorry!
here you are ..
https://www.faranux.com/product/single-channel-5v-relay-module-com41/

**Comment by robespo72** [comment]
Hi guys! When you install tasmota on a wemos (or other soc) the default wifipower is 17.0. However, on some forums I read that for the Wemos D1 Mini it is better not to go beyond the threshold of 8.0. What is your experience in this regard?

**Comment by Jason2866** [comment]
Lowering WiFi power is made to reduce the current draw. This may workaround the underlying real problem of a too weak LDO. Most WemosD1 mini are clones and does not have a 500mA type. When replacing the too weak LDO against a 500mA type, such workarounds are not needed anymore.

**Comment by joba-1** [comment]
I‘m sure that’s reason # 1

Others are 1) security (device cannot be hijacked from far away while in AP mode)
and 2) being nice to your neighbours or your other devices. If the range doubles the area where no one can send at the same time quadruples.

So, win-win-win situation. And still I am often too lazy to find out the best setting :D - should be automated…

**Reply by arendst** [reply]
Set wifipower to zero and it will be automatically set based on rssi.

**Reply by Jason2866** [reply]
Yes, testing and finding lowest tx power where a device is working well is a good idea.
This minimizes negative side effects. BUT a device should work always correctly with the specifications from the datasheet / reference design.
